zingolib/src/wallet/traits.rs

```diff
@@ -1423,9 +1389,22 @@ where
         writer.write_all(&self.diversifier().to_bytes())?;
 
         self.note().write(&mut writer)?;
-        writer.write_u64::<LittleEndian>(u64::from(*self.witnessed_position()))?;
-
-        writer.write_all(&self.nullifier().to_bytes())?;
+        writer.write_u64::<LittleEndian>(u64::from(self.witnessed_position().ok_or(
+            io::Error::new(
+                io::ErrorKind::InvalidData,
+                "Tried to write note with unknown position",
```

Comment:
Let's make this more explicit.   `position`..   of what?

Is it a `funding-value-commitment-position`?   Then let's call it exactly that everywhere.   Suddenly, if there's any other use of "position" it will be obvious, and everywhere it's actually a `funding-value-commitment-position` it will be extremely clear, that that's the "position" we mean.   Maybe also `witnessed_position` becomes:  `witnessed_funding_value_commitment_position`...

If that's too much of a mouthful, then I would be fine with a well documented and publicized nym/acronym:

`fundivalcompos`...